src/main/java/wolox/training/security/CustomAuthenticationProvider.java

```diff
@@ -0,0 +1,39 @@
+package wolox.training.security;
+
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+import wolox.training.repositories.UserRepository;
+
+@Component
+public class CustomAuthenticationProvider implements AuthenticationProvider {
+
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Override
+    public Authentication authenticate(Authentication authentication)
+            throws AuthenticationException {
+
+        String name = authentication.getName();
+        String password = authentication.getCredentials().toString();
+        userRepository.findUserByUsername(name).filter(user -> passwordEncoder.matches(password, password))
+        .orElseThrow(()-> new BadCredentialsException("Incorrect Credential"));
```

Comment:
This error could be a constant, inside the file "ConstantsErrors"

Reply:
I put this message in constant error